command/help.go

```diff
@@ -0,0 +1,117 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+func rootDisplayCommandTypoHelp(command *cobra.Command, args []string) {
+	if command != RootCmd {
+		// Display helpful error message in case subcommand name was mistyped.
+		// This matches Cobra's behavior for root command, which Cobra
+		// confusingly doesn't apply to nested commands.
+		if command.Parent() == RootCmd && len(args) >= 2 {
+			if command.SuggestionsMinimumDistance <= 0 {
+				command.SuggestionsMinimumDistance = 2
+			}
+			candidates := command.SuggestionsFor(args[1])
+
+			errOut := command.OutOrStderr()
+			fmt.Fprintf(errOut, "unknown command %q for %q\n", args[1], "gh "+args[0])
+
+			if len(candidates) > 0 {
+				fmt.Fprint(errOut, "\nDid you mean this?\n")
+				for _, c := range candidates {
+					fmt.Fprintf(errOut, "\t%s\n", c)
+				}
+				fmt.Fprint(errOut, "\n")
+			}
+
+			oldOut := command.OutOrStdout()
+			command.SetOut(errOut)
+			defer command.SetOut(oldOut)
```

Comment:
The parent function will terminate after this line, so this `defer` is now moot since the expression will be executed immediately.

I think the `command.SetOut(errOut)` hack was here because we used to invoke `cobraDefaultHelpFunc` after this, and I wanted to force the error output to go to stderr instead of stdout. The general recipe should be: help text should go to stdout only if it's explicitly requested, otherwise it should go to stderr if it's part of an error message. Now that we control the entire help function ourselves, we perhaps don't need this hack anymore?